python_modules/dagster/dagster/_core/host_representation/external_data.py

```diff
@@ -912,7 +920,10 @@ def __new__(
             ),
             configured_values=dict(
                 check.mapping_param(
-                    configured_values, "configured_values", key_type=str, value_type=str
+                    configured_values,
+                    "configured_values",
+                    key_type=str,
+                    value_type=(str, ExternalResourceConfigEnvVar),
```

Comment:
If I am reading this correct, we typehint as `Dict[str, ExternalResourceValue]` but check `value_type=(str, ExternalResourceConfigEnvVar)` for backwards compat purposes.

Definitely threw me for a second because I thought you were committing a breaking change here. I'd prefer an approach with an explicit serializer here.

Reply:
Hmm, could you elaborate on what you are looking for here? Right now the `configured_values` dict can take either strings (representing normal values) or `ExternalResourceConfigEnvVar` (representing env vars) with `ExternalResourceValue` as an alias for the union of the two.

Would you prefer that we don't use raw `str` for normal values?

Reply:
Oh I see I was confused. Also my bad on the slow turnaround here.